src/core/vector.pmt

```diff
@@ -898,7 +898,86 @@ long int FUNCTION(igraph_vector, qsort_ind)(TYPE(igraph_vector) *v,
     return 0;
 }
 
-#endif
+/**
+ * \function igraph_vector_lex_cmp
+ * \brief Lexicographical comparison of two vectors.
+ *
```

Comment:
In order to avoid any ambiguity, it will be useful to spell out that if all elements match, but the vectors have different sizes, then the shorter one comes first.